bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -675,5 +723,226 @@ private void validateSQL(String sql) throws TskCoreException {
+			}
+		}
+
+		/**
+		 * Sets the value at the given parameter index to the given value.
+		 *
+		 * @param parameterIndex The index.
+		 * @param x              The value to set at that index.
+		 *
+		 * @throws TskCoreException
+		 */
+		public void setString(int parameterIndex, String x) throws TskCoreException {
+			try {
+				preparedStatement.setString(parameterIndex, x);
+			} catch (SQLException ex) {
+				throw new TskCoreException(MessageFormat.format("There was an error setting the value at index: {0} to {1}", parameterIndex, x), ex);
+			}
+		}
+
+		/**
+		 * Sets the value at the given parameter index to the given value.
+		 *
+		 * @param parameterIndex The index.
+		 * @param x              The value to set at that index.
+		 *
+		 * @throws TskCoreException
+		 */
+		public void setDate(int parameterIndex, Date x) throws TskCoreException {
+			try {
+				preparedStatement.setDate(parameterIndex, x);
+			} catch (SQLException ex) {
+				throw new TskCoreException(MessageFormat.format("There was an error setting the value at index: {0} to {1}", parameterIndex, x), ex);
+			}
+		}
+
+		/**
+		 * Sets the value at the given parameter index to the given value.
+		 *
+		 * @param parameterIndex The index.
+		 * @param x              The value to set at that index.
+		 *
+		 * @throws TskCoreException
+		 */
+		public void setTime(int parameterIndex, Time x) throws TskCoreException {
+			try {
+				preparedStatement.setTime(parameterIndex, x);
+			} catch (SQLException ex) {
+				throw new TskCoreException(MessageFormat.format("There was an error setting the value at index: {0} to {1}", parameterIndex, x), ex);
+			}
+		}
+
+		/**
+		 * Sets the value at the given parameter index to the given value.
+		 *
+		 * @param parameterIndex The index.
+		 * @param x              The value to set at that index.
+		 *
+		 * @throws TskCoreException
+		 */
+		public void setTimestamp(int parameterIndex, Timestamp x) throws TskCoreException {
+			try {
+				preparedStatement.setTimestamp(parameterIndex, x);
+			} catch (SQLException ex) {
+				throw new TskCoreException(MessageFormat.format("There was an error setting the value at index: {0} to {1}", parameterIndex, x), ex);
+			}
+		}
+
+		/**
+		 * Sets the value at the given parameter index to the given value. The
+		 * sql type is determined in the same manner as
+		 * java.sql.PreparedStatement.setObject.
+		 *
+		 * @param parameterIndex The index.
+		 * @param x              The value to set at that index.
+		 *
+		 * @throws TskCoreException
+		 */
+		public void setObject(int parameterIndex, Object x) throws TskCoreException {
+			try {
+				preparedStatement.setObject(parameterIndex, x);
+			} catch (SQLException ex) {
+				throw new TskCoreException(MessageFormat.format("There was an error setting the value at index: {0} to {1}", parameterIndex, x), ex);
+			}
+		}
+
+		@Override
+		public void close() throws SQLException {
+			connection.close();
+			preparedStatement.close();
```

Comment:
Doesn't closing the JDBC connection close the JDBC prepared statement? In any case, if we are going to close both explicitly, I think that we need to close the prepared statement first.